### DISCUSSION TITLE
Error message "PKIX path building failed" and unable to auth on Org

### DISCUSSION THREAD

**Discussion by fmertins** [forcedotcom/dataloader#800] [Q&A] [answered]
Hey there, hope you're doing well.
We're trying to overcome the following error:

![image](https://github.com/forcedotcom/dataloader/assets/2380959/e3364851-e4c4-4dd6-a925-c4b307c80d28)

But no luck, after searching for a few posts and forums.
It's happening only on a specific sandbox Org; for example, when we run the same process on another sandbox, it works properly.

Any idea, or suggestions about what could be causing that error?
Thank you in advance!

**Comment by ashitsalesforce** [comment]
Hi @fmertins , have you looked at the following posts:
- https://developer.salesforce.com/forums/?id=9062I000000IXNgQAO
- https://stackoverflow.com/questions/21076179/pkix-path-building-failed-and-unable-to-find-valid-certification-path-to-requ
- https://github.com/forcedotcom/dataloader/issues/641#issuecomment-1510417459

In a nutshell, either your SB org or your proxy service that sits between data loader and SB org presents a cert to data loader that is not verified by a Certificate Authority. The fix is to add that cert to the default keystore for your JRE on your deskthop. A workaround is to switch to browser-based OAuth login (it is supported in the batch mode too), which will make your process scripts interactive - they will prompt you for login through a browser window before proceeding.

**Comment by fmertins** [comment, marked as answer]
Hey, thank you, today we realized the cause of the error: we were using that "myorg.sandbox.lightning.force.com" URL pattern but instead, it has to be that "myorg.sandbox.my.salesforce.com" variation. That goes on the DL's properties file, for example directives like:

- sfdc.oauth.Sandbox.server
- sfdc.endpoint

Interestingly, if we ping both URL patterns, they both seem to point to "dfw.edge2.salesforce.com"; not the very same IP, but very similar ones. They both work just fine in terms of opening them on the browser, but it looks like there's a difference when DL will use them. Does make any sense?

(On the browser, "myorg.sandbox.my.salesforce.com" ends up being redirected to "myorg.sandbox.lightning.force.com")

**Comment by ashitsalesforce** [comment]
Hi @fmertins , it is an interesting find indeed. I tried by setting `sfdc.oauth.Production.server=https://<myorg>.lightning.force.com` URL and ran into the following error before logging in:
```
ERROR [main] ui.OAuthLoginFromBrowserFlow open (OAuthLoginFromBrowserFlow.java:78) - No content to map due to end-of-input
 at [Source: (ByteArrayInputStream); line: 1, column: 0]
```

So, even storing the cert into JRE's keystore would likely not have helped in your case. We will investigate this further and see why `lightning.force.com` URLs are not working with Data Loader. Meanwhile, good to hear that switching to the `sandbox.my.salesforce.com` URL worked. One possible fix to explore would be to convert all URL patterns of type `lightning.force.com` to `my.salesforce.com`.

**Comment by ashitsalesforce** [comment]
PR #804 contains the fix to retry browser-based OAuth login by using `my.salesforce.com` suffix and eventually `https://login.salesforce.com` if initial attempt fails.

**Comment by fmertins** [comment]
Hello, thank you again for the prompt support and attention here, all the best!
